src/Embeddings/VectorStores/Qdrant/QdrantVectorStore.php

```diff
@@ -26,11 +27,31 @@ class QdrantVectorStore extends VectorStoreBase
 
     public Qdrant $client;
 
-    public function __construct(Config $config, private string $collectionName)
-    {
+    public function __construct(
+        Config $config,
+        private string $collectionName,
+        private ?string $vectorName = self::QDRANT_OPENAI_VECTOR_NAME,
+        private string $distance = VectorParams::DISTANCE_COSINE,
+    ) {
         $this->client = new Qdrant(new Transport(new Client(), $config));
     }
 
+    public function setVectorName(?string $vectorName): void
+    {
+        $this->vectorName = $vectorName;
+    }
+
+    /**
+     * @param  $distance  string [Cosine, Euclid, Dot]
+     */
+    public function setDistance(string $distance): void
+    {
+        if (in_array($distance, [VectorParams::DISTANCE_COSINE, VectorParams::DISTANCE_DOT, VectorParams::DISTANCE_EUCLID])) {
```

Comment:
shouldn't it be !in_array ?

Reply:
yes this was, this is corrected & tests added